app/controllers/api/contacts_controller.rb

```diff
@@ -0,0 +1,34 @@
+class Api::ContactsController < ApplicationController
+  def index
+    if params[:select_all] == 'true'
+      select_all_contacts
+    else
+      page = params[:page] || AppConstants::DEFAULT_PAGE
+      limit = params[:limit] || AppConstants::DEFAULT_CONTACTS_PER_PAGE
+
+      if params[:q].blank?
+        contacts = Contact.page(page).per(limit)
+      else
+        q = Contact.ransack(params[:q])
+        contacts = q.result.page(page).per(limit)
+      end
```

Comment:
```suggestion
      q = Contact.ransack(params[:q])
      contacts = q.result.page(page).per(limit)
```

Try if this works, (ransack should handle the blank `params[:q]`)